### ISSUE TITLE
`GLOB` function is unparsable

### ISSUE THREAD

**Issue by rvnkcode** [sqlfluff/sqlfluff#5717]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I've got following error.

> file_name|2 col 1 error| Line 2, Position 1: Found unparsable section: 'CREATE TABLE IF NOT EXISTS task (\n    id...'

### Expected Behaviour

The code is parsed.

### Observed Behaviour

When I run my SQL, it works just fine so I think linter should be able to parse `GLOB` function.

### How to reproduce

```sql
CREATE TABLE IF NOT EXISTS task (
    id TEXT PRIMARY KEY CHECK (length(id) = 15),
    priority TEXT CHECK (priority GLOB '[A-Z]'),
    title TEXT NOT NULL,
    note TEXT,
    created_at DATETIME NOT NULL DEFAULT current_timestamp,
    updated_at DATETIME NOT NULL DEFAULT current_timestamp
);
```

If I delete `    priority TEXT CHECK (priority GLOB '[A-Z]'),` line, linter works just fine.

### Dialect

sqlite

### Version

3.0.3

### Configuration

[sqlfluff]
dialect = sqlite

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
